src/libecalc/core/ecalc.py

```diff
@@ -16,12 +16,14 @@
 from libecalc.core.consumers.factory import create_consumer
 from libecalc.core.consumers.generator_set import Genset
 from libecalc.core.consumers.legacy_consumer.component import Consumer
-from libecalc.core.consumers.venting_emitter import VentingEmitter
 from libecalc.core.models.fuel import FuelModel
 from libecalc.core.result import ComponentResult, EcalcModelResult
 from libecalc.core.result.emission import EmissionResult
 from libecalc.dto.component_graph import ComponentGraph
 from libecalc.dto.types import ConsumptionType
+from libecalc.presentation.yaml.yaml_types.emitters.yaml_venting_emitter import (
```

Comment:
so; I see that the dto is skipped, which is good, but in that case the yaml should be converted to the domain object directly, instead of referring to the yaml object in core... as a rule, core should only depend on core (and common), not presentation, application or infrastructure layers...

Reply:
When I removed the venting emitter from core, the remaining logic was put in the presentation layer (yaml class). Basically only evaluating expressions and handling stream days/calendar days (`get_emission_rate`). It may be possible to argue that there is no domain specific logic for venting emitters (except for rate type check). We may need a domain object, regardless of how much it does?

Reply:
Good point!

I think yes. 3 arguments from top of my head; 1. we may want to use the domain object directly, when using the libeCalc as a (library) API. 2. the domain may grow, and it should grow naturally from the e.g. basic logic we initially have, instead of having to change on a lot of the rest of the structure. 3. when the dependencies are "tangled", it gets much more difficult to change the code, side effects etc. My 3 cents :p